sdk/src/Services/APIGateway/Generated/Model/Internal/MarshallTransformations/GetSdkRequestMarshaller.cs

```diff
@@ -67,11 +67,12 @@ public IRequest Marshall(GetSdkRequest publicRequest)
             if (!publicRequest.IsSetStageName())
                 throw new AmazonAPIGatewayException("Request object does not have required field StageName set");
             request.AddPathResource("{stage_name}", StringUtils.FromString(publicRequest.StageName));
-            
             if (publicRequest.IsSetParameters())
             {
                 foreach(var kvp in publicRequest.Parameters)
                 {
+                    if(request.Parameters.ContainsKey(kvp.Key))
```

Comment:
I am confused how the key would exist here in this case because Parameters is already a Dictionary. Is this the case where it could have already been a set property on the request that was already added? If that situation doesn't exist though it seems like we shouldn't be trying to remove keys.

Reply:
If you look at `PutRestApiRequestMarshaller`, if Mode is set the key will be added, but later down the line when going through the map of query params, it will try to add a key that is already there if that remove statement is not there. Having said that though, I just realized from reading the smithy doc that I implemented this backwards :( . I should be disregarding the value set by the query param of maps and taking the value of the httpQuery trait. I will need to double check the protocol test b/c it did pass for some reason